k8s/factory_test.go

```diff
@@ -0,0 +1,20 @@
+package k8s
+
+import "k8s.io/client-go/kubernetes/fake"
+
+func mockFactory() Factory {
```

Comment:
What is `Factory`? Can we make the name more specific please?

Reply:
Any suggestions?

Reply:
Yes. Prefix it with what it's a factory for?